src/main/java/frc/robot/subsystems/TurretSubsystem.java

```diff
@@ -0,0 +1,109 @@
+  private CANcoder m_encoder;
+  private double turretPosition;
+  private double turretVelocity;
+  private DebugEntry<Double> turretPositionEntry =
+      new DebugEntry<Double>(turretPosition, "Position", this);
+  private DebugEntry<Double> turretVelocityEntry =
+      new DebugEntry<Double>(turretVelocity, "Velocity", this);
+
+  public TurretSubsystem() {
+    turretMotor = new CANSparkMax(Constants.TurretConstants.TURRET_MOTOR_ID, MotorType.kBrushless);
+    turretMotor.restoreFactoryDefaults();
+    turretMotor.setSmartCurrentLimit(40);
+
+    turretMotor.enableSoftLimit(CANSparkMax.SoftLimitDirection.kForward, true);
+    turretMotor.enableSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, true);
+
+    turretMotor.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, 1);
+    turretMotor.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, -1);
+
+    // initialze PID controller and encoder objects
+    turretPIDController =
+        new PIDController(
+            Constants.TurretConstants.TURRET_KP,
+            Constants.TurretConstants.TURRET_KI,
+            Constants.TurretConstants.TURRET_KD);
+    m_encoder = new CANcoder(Constants.TurretConstants.CANcoder_ID);
+
+    // set PID coefficients
+    turretPIDController.setP(Constants.TurretConstants.TURRET_KP);
+    turretPIDController.setI(Constants.TurretConstants.TURRET_KI);
+    turretPIDController.setD(Constants.TurretConstants.TURRET_KD);
+    turretPIDController.setIZone(Constants.TurretConstants.TURRET_KIZ);
+  }
+
+  public void stopTurret() {
+    turretMotor.stopMotor();
+  }
+
+  public void setTurretPos(double setpoint) {
+    turretMotor.set(turretPIDController.calculate(turretPosition, setpoint));
+  }
+
+  public void zeroTurretEncoder() {
+    m_encoder.setPosition(0.0);
+  }
+
+  public void updateTurretPosition() {
+    turretPosition =
+        (Math.toRadians((m_encoder.getPosition().getValueAsDouble()) * 360))
+            * Constants.TurretConstants.CONVERSION_FACTOR;
+  }
+
+  public void lockOntoTag(double rotationFromTag) {
+    turretMotor.set(turretPIDController.calculate(rotationFromTag, 0.0));
+  }
+
+  public double getTurretPos() {
+    return turretPosition; // returns the absolute encoder position in radians
+  }
+
+  public double getTurretVel() {
+    return turretVelocity;
+  }
+
+  public Command LockTurret() {
+    return run(() -> lockOntoTag(LimelightHelpers.getTX("")));
+  }
+
+  @Override
+  public void periodic() {
+    updateTurretPosition();
+    turretVelocity =
+        (m_encoder.getVelocity().getValueAsDouble())
+            * 60; // changing from rotations per second to rotations per minute or rpm
+    turretPositionEntry.log(turretPosition);
+    turretVelocityEntry.log(turretVelocity);
+  }
+
+  public double turretFromOdometry(
+      Pose2d robotPos) { // Doesn't work right now but will be used later
+    return Math.atan(robotPos.getY() / robotPos.getX()) + robotPos.getRotation().getDegrees();
+  }
```

Comment:
No? We need the X and Y because we're having the turret track a target just from robot position without any AprilTags. This wouldn't do anything without the X and Y.

Reply:
Currently this tracks the line from the field origin through the robot's position, which does not lead to a static position/target as the robot moves. 

turretFromOdometry() is also not a descriptive method name for the intent described.